File: ExampleObjc/ObjcSwiftTest/ViewController.m
//
//  ViewController.m
//  ObjcSwiftTest
//
//  Created by boska on 23/05/2017.
//  Copyright © 2017 boska. All rights reserved.
//

#import "ViewController.h"
#import <AVFoundation/AVFoundation.h>


@import ViscoveryADSDK;

@interface VideoView : UIView
  @end
@implementation VideoView
+ (Class) layerClass {
  return [AVPlayerLayer class];
}
  @end

@interface ViewController ()
@property(nonatomic,weak) IBOutlet VideoView *videoView;
@property(nonatomic,strong) AVPlayer *contentPlayer;
@property(nonatomic,strong) AdsManager *adsManager;
@end


@implementation ViewController

- (void)viewDidLoad {
  [super viewDidLoad];
  
  NSURL *url = [[NSURL alloc]initWithString:@"http://viscovery-vsp-dev.s3.amazonaws.com/sdkdemo/Videos/Mobile%20App_Demo%20Video%20(540p).mp4"];
  
  self.contentPlayer = [AVPlayer playerWithURL:url];
  ((AVPlayerLayer *)self.videoView.layer).player = self.contentPlayer;
  self.adsManager = [[AdsManager alloc] initWithPlayer:self.contentPlayer videoView:self.videoView outstreamContainerView:nil];
  [self.adsManager requestAdsWithVideoURL:nil];
  
  //[self.adsManager requestAdsWithVideoURL:nil];
}

@end

File: Example/ViscoveryADSDK/ViewController.swift
//
//  ViewController.swift
//  ViscoveryADSDK
//
//  Created by boska on 05/19/2017.
//  Copyright (c) 2017 boska. All rights reserved.
//

import UIKit
import ViscoveryADSDK
import AVFoundation

class ViewController: UIViewController {
  @IBOutlet weak var videoContainer: VideoView!
  @IBOutlet weak var outstreamContainer: UIView!
  var contentPlayer: AVPlayer?
  var adsManager: AdsManager!
  override func viewDidLoad() {
    super.viewDidLoad()
    
    guard let contentURL = URL(string: "http://viscovery-vsp-dev.s3.amazonaws.com/sdkdemo/Videos/Mobile%20App_Demo%20Video%20(540p).mp4") else { return }
    
    let url = getFileURLFromDocumentFirst() ?? contentURL
    
    contentPlayer = AVPlayer(url: url)
    videoContainer.player = contentPlayer
    adsManager = AdsManager(player: contentPlayer!, videoView: videoContainer, outstreamContainerView: outstreamContainer)
    
    contentPlayer?.isMuted = true
    
    adsManager.requestAds()
    
    //you can adjust y-pos of the instream ads (0.0~50.0) in points
    //adsManager.instreamOffset = 20.0
    
  }
  func getFileURLFromDocumentFirst() -> URL? {
    let documentsUrl =  FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first!
    
    do {
      let directoryContents = try FileManager.default.contentsOfDirectory(at: documentsUrl, includingPropertiesForKeys: nil, options: [])
      return directoryContents.first
    } catch _ {
      return nil
    }
  }
}
